beacon-chain/powchain/engine-api-client/v1/client.go

```diff
@@ -67,19 +67,59 @@ func New(ctx context.Context, endpoint string, opts ...Option) (*Client, error)
 	return c, nil
 }
 
-// NewPayload --
-func (*Client) NewPayload(_ context.Context, _ *pb.ExecutionPayload) (*pb.PayloadStatus, error) {
-	return nil, errors.New("unimplemented")
+// NewPayload calls the engine_newPayloadV1 method via JSON-RPC.
+func (c *Client) NewPayload(ctx context.Context, payload *pb.ExecutionPayload) (*pb.PayloadStatus, error) {
+	result := &pb.PayloadStatus{}
+	err := c.rpc.CallContext(ctx, result, NewPayloadMethod, payload)
+	return result, handleRPCError(err)
 }
 
-// ForkchoiceUpdated --
-func (*Client) ForkchoiceUpdated(
-	_ context.Context, _ *pb.ForkchoiceState, _ *pb.PayloadAttributes,
+// ForkchoiceUpdated calls the engine_forkchoiceUpdatedV1 method via JSON-RPC.
+func (c *Client) ForkchoiceUpdated(
+	ctx context.Context, state *pb.ForkchoiceState, attrs *pb.PayloadAttributes,
 ) (*ForkchoiceUpdatedResponse, error) {
-	return nil, errors.New("unimplemented")
+	result := &ForkchoiceUpdatedResponse{}
+	err := c.rpc.CallContext(ctx, result, ForkchoiceUpdatedMethod, state, attrs)
+	return result, handleRPCError(err)
 }
 
-// GetPayload --
-func (*Client) GetPayload(_ context.Context, _ [8]byte) (*pb.ExecutionPayload, error) {
-	return nil, errors.New("unimplemented")
+// GetPayload calls the engine_getPayloadV1 method via JSON-RPC.
+func (c *Client) GetPayload(ctx context.Context, payloadId [8]byte) (*pb.ExecutionPayload, error) {
```

Comment:
Should it be `pb.HexBytes ` ?

```suggestion
func (c *Client) GetPayload(ctx context.Context, payloadId pb.HexBytes ) (*pb.ExecutionPayload, error) {
```

Reply:
Got it